src/generators/VueGenerator.js

```diff
@@ -69,128 +68,170 @@ export default class extends BaseGenerator {
-      "store/modules/%s/update/index.js",
-      "store/modules/%s/update/mutation_types.js",
-      "store/modules/%s/update/mutations.js",
+      "stores/%s/create.ts",
+      "stores/%s/delete.ts",
+      "stores/%s/list.ts",
+      "stores/%s/show.ts",
+      "stores/%s/update.ts",
+
+      // views
+      "views/%s/CreateView.vue",
+      "views/%s/ListView.vue",
+      "views/%s/ShowView.vue",
+      "views/%s/UpdateView.vue",
 
       // components
-      "components/%s/Create.vue",
-      "components/%s/Form.vue",
-      "components/%s/List.vue",
-      "components/%s/Update.vue",
-      "components/%s/Show.vue",
+      "components/%s/EntityCreate.vue",
+      "components/%s/EntityForm.vue",
+      "components/%s/EntityList.vue",
+      "components/%s/EntityUpdate.vue",
+      "components/%s/EntityShow.vue",
 
       // routes
-      "router/%s.js",
+      "router/%s.ts",
     ].forEach((pattern) =>
       this.createFileFromPattern(pattern, dir, lc, context)
     );
 
-    for (const file of ["mixins/ItemWatcher.js", "mixins/ListWatcher.js"]) {
-      this.createFile(file, `${dir}/${file}`);
-    }
+    // common components
+    this.createFile(
+      "components/common/FormRepeater.vue",
+      `${dir}/components/common/FormRepeater.vue`,
+      context,
+      false
+    );
+
+    // composables
+    this.createFile(
+      "composables/mercureItem.ts",
+      `${dir}/composables/mercureItem.ts`,
+      {},
+      false
+    );
+    this.createFile(
+      "composables/mercureList.ts",
+      `${dir}/composables/mercureList.ts`,
+      {},
+      false
+    );
 
     // error
     this.createFile(
-      "error/SubmissionError.js",
-      `${dir}/error/SubmissionError.js`,
+      "error/SubmissionError.ts",
+      `${dir}/error/SubmissionError.ts`,
       context,
       false
     );
 
-    this.createEntrypoint(api.entrypoint, `${dir}/config/entrypoint.js`);
+    this.createEntrypoint(api.entrypoint, `${dir}/config/entrypoint.ts`);
+    this.createFile("utils/date.ts", `${dir}/utils/date.ts`, {}, false);
+    this.createFile(
+      "utils/fetch.ts",
+      `${dir}/utils/fetch.ts`,
+      { hydraPrefix: this.hydraPrefix },
+      false
+    );
     this.createFile(
-      "utils/fetch.js",
-      `${dir}/utils/fetch.js`,
+      "utils/hydra.ts",
+      `${dir}/utils/hydra.ts`,
       { hydraPrefix: this.hydraPrefix },
       false
     );
     this.createFile(
-      "utils/hydra.js",
-      `${dir}/utils/hydra.js`,
+      "utils/types.ts",
+      `${dir}/utils/types.ts`,
       { hydraPrefix: this.hydraPrefix },
       false
     );
-    this.createFile("utils/mercure.js", `${dir}/utils/mercure.js`);
+    this.createFile("utils/mercure.js", `${dir}/utils/mercure.ts`);
+  }
+
+  parseFields(resource) {
+    const fields = [
+      ...resource.writableFields,
+      ...resource.readableFields,
+    ].reduce((list, field) => {
+      if (list[field.name]) {
+        return list;
+      }
+
+      const isReferences = Boolean(
```

Comment:
Why using a `Boolean`?

Reply:
It is to convert value as a boolean as it can be null